lib/std/os/windows.zig

```diff
@@ -422,6 +422,10 @@ pub fn RtlGenRandom(output: []u8) RtlGenRandomError!void {
         const to_read: ULONG = @min(buff.len, max_read_size);
 
         if (advapi32.RtlGenRandom(buff.ptr, to_read) == 0) {
+            // This function has been seen to fail in the wild with error code 38 (Reached the end of the file),
+            // but that happened while the system was running out of system resources and that failure
+            // has not been reproduced. Additionally, RtlGenRandom internally calls ProcessPrng which
+            // is documented to not be able to fail, so any error is truly unexpected.
```

Comment:
`unexpectedError` is for errors that are truly unexpected, and will possibly print a message to the programmer like "please report this to the zig bug tracker". This error code 38 which has been observed is not truly unexpected anymore.

It needs an error code other than `error.Unexpected`. I suggest mapping it to `error.SystemResources`.

Reply:
You're not hallucinating, I changed my mind. I appreciated the comment but also had misgivings about it and days later I have a more clear suggestion (see the comment I just added to that issue).